Sources/XiEditor/EditViewController.swift

```diff
@@ -370,10 +370,9 @@ class EditViewController: NSViewController, EditViewDataSource, FindDelegate, Sc
         DispatchQueue.main.async { [weak self] in
             self?.document.updateChangeCount(hasNoUnsavedChanges ? .changeCleared : .changeDone)
             
-            // If the document's edited status has changed then display as such in the window and tab titles.
-            if let window = self?.document.windowControllers[0] as? XiWindowController,
-                window.editedStatus != !hasNoUnsavedChanges {
-                window.editedStatus = !hasNoUnsavedChanges
+            // Display the document's edited status in the window and tab titles.
```

Comment:
is this still accurate? my understanding is we don't show the status in the window title.

Reply:
The tab and window title mimic one another.  So upon switching to a new tab, the window title changes as well.  So the edited indicator will also be displayed (or not displayed) in both places.

Reply:
ah okay I understand, we always set this here and then basically filter in the `didSet`? 

My only nit then would maybe be the name `showEditedIndicator`.

Thinking about this more, can't determine whether there are unsaved changes from `document.hasUnautosavedChanges`? This should always be up to date, so we can maybe just check the documents count and this value in the `windowTitle` override?

Reply:
That last suggestion does work well!  When I was originally starting this I couldn't seem to get `document.hasUnautosavedChanges` to be the expected result; but in hindsight it was likely an error on my part while testing.  A new commit removes the `showEditedIndicator` state variable in favor of the logic being moved into the `windowTitle(forDocumentDisplayName:)` method.  I believe this is much more simplified and doesn't rely on a state variable (which I don't generally like using in these circumstances).